### ISSUE TITLE
Integrated graphic card crash on AsyncWorldInit

### ISSUE THREAD

**Issue by tomedi** [REGoth-project/REGoth#195]
Integrated graphic cards crash when executing newworld.zen.

**Comment by tomedi** [comment]
Latest crashlog: 

```
*******************************************************************************
*                                                                             *
*                        Exception Analysis                                   *
*                                                                             *
*******************************************************************************

*** ERROR: Symbol file could not be found.  Defaulted to export symbols for OWExplorer.dll - 
*** ERROR: Symbol file could not be found.  Defaulted to export symbols for ig75icd32.dll - 
*** WARNING: Unable to verify checksum for OpenAL32_RelWithDebugInfo.dll

DUMP_CLASS: 2

DUMP_QUALIFIER: 400

CONTEXT:  (.ecxr)
eax=00000000 ebx=0000f34c ecx=00242c38 edx=00002000 esi=00000068 edi=0b6bdc9c
eip=00eac7b0 esp=0b6bdc80 ebp=0b6bdc84 iopl=0         nv up ei pl zr na pe nc
cs=0023  ss=002b  ds=002b  es=002b  fs=0053  gs=002b             efl=00000246
REGoth!bx::debugBreak:
00eac7b0 cc              int     3
Resetting default scope

FAULTING_IP: 
REGoth!bx::debugBreak+0 [c:\users\berto\documents\github\regoth_orig\lib\bgfx-cmake\bx\src\debug.cpp @ 30]
00eac7b0 cc              int     3

EXCEPTION_RECORD:  (.exr -1)
ExceptionAddress: 00eac7b0 (REGoth!bx::debugBreak)
   ExceptionCode: 80000003 (Break instruction exception)
  ExceptionFlags: 00000000
NumberParameters: 1
   Parameter[0]: 00000000

DEFAULT_BUCKET_ID:  STATUS_BREAKPOINT

PROCESS_NAME:  REGoth.exe

ERROR_CODE: (NTSTATUS) 0x80000003 - {AUSNAHME}  Haltepunkt  Im Quellprogramm wurde ein Haltepunkt erreicht.

EXCEPTION_CODE: (HRESULT) 0x80000003 (2147483651) - Mindestens ein Argument ist ung ltig.

EXCEPTION_CODE_STR:  80000003

EXCEPTION_PARAMETER1:  00000000

WATSON_BKT_PROCSTAMP:  59493965

WATSON_BKT_MODULE:  REGoth.exe

WATSON_BKT_MODSTAMP:  59493965

WATSON_BKT_MODOFFSET:  23c7b0

BUILD_VERSION_STRING:  10.0.15063.296 (WinBuild.160101.0800)

MODLIST_WITH_TSCHKSUM_HASH:  f12a067a62158708860f8de9db2d7f32105fdac6

MODLIST_SHA1_HASH:  fe740932bf15fc10b1d2aa9a160dbc979c9f98da

NTGLOBALFLAG:  0

APPLICATION_VERIFIER_FLAGS:  0

PRODUCT_TYPE:  1

SUITE_MASK:  784

DUMP_FLAGS:  8000c07

DUMP_TYPE:  0

ANALYSIS_SESSION_HOST:  TOMEDI

ANALYSIS_SESSION_TIME:  06-20-2017 17:17:35.0695

ANALYSIS_VERSION: 10.0.14321.1024 amd64fre

THREAD_ATTRIBUTES: 
OS_LOCALE:  DEA

PROBLEM_CLASSES: 




    Tid    [0x0]
    Frame  [0x00]
    String [STATUS_BREAKPOINT]
    Data Bucketing


BUGCHECK_STR:  STATUS_BREAKPOINT

LAST_CONTROL_TRANSFER:  from 00e78eef to 00eac7b0

STACK_TEXT:  
0b6bdc7c 00e78eef 0b6bfca0 00e78fa7 00000000 REGoth!bx::debugBreak
0b6bdc84 00e78fa7 00000000 0b6bdc9c 00000505 REGoth!bgfx::CallbackStub::fatal+0xf
0b6bfca0 00e6d501 00000000 010c05d8 00000505 REGoth!bgfx::fatal+0x87
0b6bfce0 00e71bd5 0000f34c 551e7820 74ba0006 REGoth!bgfx::gl::VertexBufferGL::create+0x141
0b6bfcf8 00e93db3 00001101 5478dbc8 74ba0006 REGoth!bgfx::gl::RendererContextGL::createVertexBuffer+0x25
0b6bfe84 00e935d4 0b120e34 084f2140 0b6bfeb8 REGoth!bgfx::Context::rendererExecCommands+0x343
0b6bfe94 00e9372c ffffffff 00c7a69b 00e9385c REGoth!bgfx::Context::renderFrame+0x64
0b6bfea0 00e9385c 00eaca6e 084f2040 00c7a69b REGoth!bgfx::renderFrame+0x6c
0b6bfea4 00eaca6e 084f2040 00c7a69b 00eaf3fb REGoth!bgfx::Context::renderThread+0x1c
0b6bfeb0 00eaf3fb 0b6bfecc 75438744 084f2140 REGoth!bx::Thread::entry+0x1e
0b6bfeb8 75438744 084f2140 75438720 38a95ed9 REGoth!bx::ThreadInternal::threadFunc+0xb
0b6bfecc 77c3587d 084f2140 c069df05 00000000 kernel32!BaseThreadInitThunk+0x24
0b6bff14 77c3584d ffffffff 77c56331 00000000 ntdll!__RtlUserThreadStart+0x2f
0b6bff24 00000000 00c7a69b 084f2140 00000000 ntdll!_RtlUserThreadStart+0x1b


THREAD_SHA1_HASH_MOD_FUNC:  5413718464c79a9af7e95f0e93b41952afb523d5

THREAD_SHA1_HASH_MOD_FUNC_OFFSET:  e1360827ea608b9186234cd074df132a6ec91040

THREAD_SHA1_HASH_MOD:  0184952f04e9582b07181e0e9e3e5104f526474a

FOLLOWUP_IP: 
REGoth!bx::debugBreak+0 [c:\users\berto\documents\github\regoth_orig\lib\bgfx-cmake\bx\src\debug.cpp @ 30]
00eac7b0 cc              int     3

FAULT_INSTR_CODE:  ccccc3cc

FAULTING_SOURCE_LINE:  c:\users\berto\documents\github\regoth_orig\lib\bgfx-cmake\bx\src\debug.cpp

FAULTING_SOURCE_FILE:  c:\users\berto\documents\github\regoth_orig\lib\bgfx-cmake\bx\src\debug.cpp

FAULTING_SOURCE_LINE_NUMBER:  30

SYMBOL_STACK_INDEX:  0

SYMBOL_NAME:  REGoth!bx::debugBreak+0

FOLLOWUP_NAME:  MachineOwner

MODULE_NAME: REGoth

IMAGE_NAME:  REGoth.exe

DEBUG_FLR_IMAGE_TIMESTAMP:  59493965

STACK_COMMAND:  .ecxr ; kb

BUCKET_ID:  STATUS_BREAKPOINT_REGoth!bx::debugBreak+0

PRIMARY_PROBLEM_CLASS:  STATUS_BREAKPOINT_REGoth!bx::debugBreak+0

FAILURE_EXCEPTION_CODE:  80000003

FAILURE_IMAGE_NAME:  REGoth.exe

BUCKET_ID_IMAGE_STR:  REGoth.exe

FAILURE_MODULE_NAME:  REGoth

BUCKET_ID_MODULE_STR:  REGoth

FAILURE_FUNCTION_NAME:  bx::debugBreak

BUCKET_ID_FUNCTION_STR:  bx::debugBreak

BUCKET_ID_OFFSET:  0

BUCKET_ID_MODTIMEDATESTAMP:  59493965

BUCKET_ID_MODCHECKSUM:  0

BUCKET_ID_MODVER_STR:  0.0.0.0

BUCKET_ID_PREFIX_STR:  STATUS_BREAKPOINT_

FAILURE_PROBLEM_CLASS:  STATUS_BREAKPOINT

FAILURE_SYMBOL_NAME:  REGoth.exe!bx::debugBreak

FAILURE_BUCKET_ID:  STATUS_BREAKPOINT_80000003_REGoth.exe!bx::debugBreak

TARGET_TIME:  2017-06-20T15:14:53.000Z

OSBUILD:  15063

OSSERVICEPACK:  296

SERVICEPACK_NUMBER: 0

OS_REVISION: 0

OSPLATFORM_TYPE:  x86

OSNAME:  Windows 10

OSEDITION:  Windows 10 WinNt SingleUserTS Personal

USER_LCID:  0

OSBUILD_TIMESTAMP:  unknown_date

BUILDDATESTAMP_STR:  160101.0800

BUILDLAB_STR:  WinBuild

BUILDOSVER_STR:  10.0.15063.296

ANALYSIS_SESSION_ELAPSED_TIME: efa

ANALYSIS_SOURCE:  UM

FAILURE_ID_HASH_STRING:  um:status_breakpoint_80000003_regoth.exe!bx::debugbreak

FAILURE_ID_HASH:  {b2eb1f25-9d48-ca32-b6f4-b876a3c6e720}

Followup:     MachineOwner
---------


```

**Comment by tomedi** [comment]
Crashlog G1- world.zen- whenever you switch back from another world, world.zen crashes. Log: 
```
*******************************************************************************
*                                                                             *
*                        Exception Analysis                                   *
*                                                                             *
*******************************************************************************

*** ERROR: Symbol file could not be found.  Defaulted to export symbols for ig75icd32.dll - 
*** WARNING: Unable to verify checksum for OpenAL32_RelWithDebugInfo.dll

DUMP_CLASS: 2

DUMP_QUALIFIER: 400

CONTEXT:  (.ecxr)
eax=00000000 ebx=00006050 ecx=00c2a818 edx=00002000 esi=00000068 edi=0bb1d924
eip=0129c7b0 esp=0bb1d908 ebp=0bb1d90c iopl=0         nv up ei pl zr na pe nc
cs=0023  ss=002b  ds=002b  es=002b  fs=0053  gs=002b             efl=00000246
REGoth!bx::debugBreak:
0129c7b0 cc              int     3
Resetting default scope

FAULTING_IP: 
REGoth!bx::debugBreak+0 [c:\users\berto\documents\github\regoth_orig\lib\bgfx-cmake\bx\src\debug.cpp @ 30]
0129c7b0 cc              int     3

EXCEPTION_RECORD:  (.exr -1)
ExceptionAddress: 0129c7b0 (REGoth!bx::debugBreak)
   ExceptionCode: 80000003 (Break instruction exception)
  ExceptionFlags: 00000000
NumberParameters: 1
   Parameter[0]: 00000000

DEFAULT_BUCKET_ID:  STATUS_BREAKPOINT

PROCESS_NAME:  REGoth.exe

ERROR_CODE: (NTSTATUS) 0x80000003 - {AUSNAHME}  Haltepunkt  Im Quellprogramm wurde ein Haltepunkt erreicht.

EXCEPTION_CODE: (HRESULT) 0x80000003 (2147483651) - Mindestens ein Argument ist ung ltig.

EXCEPTION_CODE_STR:  80000003

EXCEPTION_PARAMETER1:  00000000

WATSON_BKT_PROCSTAMP:  59493965

WATSON_BKT_MODULE:  REGoth.exe

WATSON_BKT_MODSTAMP:  59493965

WATSON_BKT_MODOFFSET:  23c7b0

BUILD_VERSION_STRING:  10.0.15063.296 (WinBuild.160101.0800)

MODLIST_WITH_TSCHKSUM_HASH:  1e742791033462c11cbd0bd75b3be54b024754c5

MODLIST_SHA1_HASH:  8b91e8a8e3203459389472687bb4bfbb2b2155cd

NTGLOBALFLAG:  0

APPLICATION_VERIFIER_FLAGS:  0

PRODUCT_TYPE:  1

SUITE_MASK:  784

DUMP_FLAGS:  8000c07

DUMP_TYPE:  0

ANALYSIS_SESSION_HOST:  TOMEDI

ANALYSIS_SESSION_TIME:  06-20-2017 17:42:27.0876

ANALYSIS_VERSION: 10.0.14321.1024 amd64fre

THREAD_ATTRIBUTES: 
OS_LOCALE:  DEA

PROBLEM_CLASSES: 




    Tid    [0x0]
    Frame  [0x00]
    String [STATUS_BREAKPOINT]
    Data Bucketing


BUGCHECK_STR:  STATUS_BREAKPOINT

LAST_CONTROL_TRANSFER:  from 01268eef to 0129c7b0

STACK_TEXT:  
0bb1d904 01268eef 0bb1f928 01268fa7 00000000 REGoth!bx::debugBreak
0bb1d90c 01268fa7 00000000 0bb1d924 00000505 REGoth!bgfx::CallbackStub::fatal+0xf
0bb1f928 0125d501 00000000 014b05d8 00000505 REGoth!bgfx::fatal+0x87
0bb1f968 01261bd5 00006050 57fdcb20 74ba0006 REGoth!bgfx::gl::VertexBufferGL::create+0x141
0bb1f980 01283db3 000028d5 3ac7f6b0 74ba0006 REGoth!bgfx::gl::RendererContextGL::createVertexBuffer+0x25
0bb1fb0c 012835d4 09f139f4 08956140 0bb1fb40 REGoth!bgfx::Context::rendererExecCommands+0x343
0bb1fb1c 0128372c ffffffff 0106a69b 0128385c REGoth!bgfx::Context::renderFrame+0x64
0bb1fb28 0128385c 0129ca6e 08956040 0106a69b REGoth!bgfx::renderFrame+0x6c
0bb1fb2c 0129ca6e 08956040 0106a69b 0129f3fb REGoth!bgfx::Context::renderThread+0x1c
0bb1fb38 0129f3fb 0bb1fb54 75438744 08956140 REGoth!bx::Thread::entry+0x1e
0bb1fb40 75438744 08956140 75438720 f7ea2b1e REGoth!bx::ThreadInternal::threadFunc+0xb
0bb1fb54 77c3587d 08956140 f51e3acb 00000000 kernel32!BaseThreadInitThunk+0x24
0bb1fb9c 77c3584d ffffffff 77c5635b 00000000 ntdll!__RtlUserThreadStart+0x2f
0bb1fbac 00000000 0106a69b 08956140 00000000 ntdll!_RtlUserThreadStart+0x1b


THREAD_SHA1_HASH_MOD_FUNC:  5413718464c79a9af7e95f0e93b41952afb523d5

THREAD_SHA1_HASH_MOD_FUNC_OFFSET:  e1360827ea608b9186234cd074df132a6ec91040

THREAD_SHA1_HASH_MOD:  0184952f04e9582b07181e0e9e3e5104f526474a

FOLLOWUP_IP: 
REGoth!bx::debugBreak+0 [c:\users\berto\documents\github\regoth_orig\lib\bgfx-cmake\bx\src\debug.cpp @ 30]
0129c7b0 cc              int     3

FAULT_INSTR_CODE:  ccccc3cc

FAULTING_SOURCE_LINE:  c:\users\berto\documents\github\regoth_orig\lib\bgfx-cmake\bx\src\debug.cpp

FAULTING_SOURCE_FILE:  c:\users\berto\documents\github\regoth_orig\lib\bgfx-cmake\bx\src\debug.cpp

FAULTING_SOURCE_LINE_NUMBER:  30

SYMBOL_STACK_INDEX:  0

SYMBOL_NAME:  REGoth!bx::debugBreak+0

FOLLOWUP_NAME:  MachineOwner

MODULE_NAME: REGoth

IMAGE_NAME:  REGoth.exe

DEBUG_FLR_IMAGE_TIMESTAMP:  59493965

STACK_COMMAND:  .ecxr ; kb

BUCKET_ID:  STATUS_BREAKPOINT_REGoth!bx::debugBreak+0

PRIMARY_PROBLEM_CLASS:  STATUS_BREAKPOINT_REGoth!bx::debugBreak+0

FAILURE_EXCEPTION_CODE:  80000003

FAILURE_IMAGE_NAME:  REGoth.exe

BUCKET_ID_IMAGE_STR:  REGoth.exe

FAILURE_MODULE_NAME:  REGoth

BUCKET_ID_MODULE_STR:  REGoth

FAILURE_FUNCTION_NAME:  bx::debugBreak

BUCKET_ID_FUNCTION_STR:  bx::debugBreak

BUCKET_ID_OFFSET:  0

BUCKET_ID_MODTIMEDATESTAMP:  59493965

BUCKET_ID_MODCHECKSUM:  0

BUCKET_ID_MODVER_STR:  0.0.0.0

BUCKET_ID_PREFIX_STR:  STATUS_BREAKPOINT_

FAILURE_PROBLEM_CLASS:  STATUS_BREAKPOINT

FAILURE_SYMBOL_NAME:  REGoth.exe!bx::debugBreak

FAILURE_BUCKET_ID:  STATUS_BREAKPOINT_80000003_REGoth.exe!bx::debugBreak

TARGET_TIME:  2017-06-20T15:36:20.000Z

OSBUILD:  15063

OSSERVICEPACK:  296

SERVICEPACK_NUMBER: 0

OS_REVISION: 0

OSPLATFORM_TYPE:  x86

OSNAME:  Windows 10

OSEDITION:  Windows 10 WinNt SingleUserTS Personal

USER_LCID:  0

OSBUILD_TIMESTAMP:  unknown_date

BUILDDATESTAMP_STR:  160101.0800

BUILDLAB_STR:  WinBuild

BUILDOSVER_STR:  10.0.15063.296

ANALYSIS_SESSION_ELAPSED_TIME: 1ac0

ANALYSIS_SOURCE:  UM

FAILURE_ID_HASH_STRING:  um:status_breakpoint_80000003_regoth.exe!bx::debugbreak

FAILURE_ID_HASH:  {b2eb1f25-9d48-ca32-b6f4-b876a3c6e720}

Followup:     MachineOwner
---------


```

**Comment by tomedi** [comment]
Fixed with AsyncWorldInit 0.3.308
